src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs

```diff
@@ -75,5 +78,11 @@ public CompilerCacheResult(IList<IChangeToken> expirationTokens)
         /// Gets a value that determines if the view was successfully found and compiled.
         /// </summary>
         public bool Success { get; }
+
+        /// <summary>
+        /// Gets a Func to create the view.
+        /// </summary>
+        /// <remarks>This property is not available when <see cref="Success"/> is <c>false</c>.</remarks>
+        public Func<object> PageCreate { get; }
```

Comment:
@rynowak `Activator` also requires the parameterless constructor. The only new (pardon the pun) restriction I see relates to the corner case where `Activator` has sufficient access to invoke a non-`public` constructor.